src/RuntimeModule.ts

```diff
@@ -0,0 +1,43 @@
+/*
+Copyright 2022 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { EventEmitter } from "events";
+
+import { ModuleApi } from "./ModuleApi";
+
+// TODO: Type the event emitter with AnyLifecycle (extract TypedEventEmitter from js-sdk somehow?)
+// See https://github.com/matrix-org/matrix-react-sdk-module-api/issues/4
```

Comment:
Couldn't this just grow a peer-dep to js-sdk given it knows it'll have a js-sdk from react-sdk?

Reply:
Potentially. I've been trying to avoid the peer dependency style as it'll end up being a slew of warnings for developers (because we won't update the dependency very often)

Reply:
that's fine though because its a peer-dep, you could even specify the min version and say any version from there. `>=12.4.1` (replacing with the actual version we introduced TypedEventEmitter)

Reply:
until we release a breaking change of the js-sdk and then it is considered unmatched (both logically and by the semver checks - we can't assume it'll continue to compile). 

For now I'm not that concerned with typed event emitter support. Documentation can supplement it.

Reply:
Then why not add yet another layer called `matrix-js-common` or something :P?

Reply:
not sure if serious, but it's quite literally on the table for consideration.

Reply:
It'd be a good home for some things for element-desktop to re-use too, like `fetchdep.sh`

Reply:
probably for a larger conversation outside of this PR though ;)